source/Command.cpp

```diff
@@ -69,7 +69,15 @@ const Command Command::AMMO(1uL << 25, "Fleet: Toggle ammo usage");
 const Command Command::WAIT(1uL << 26, "");
 const Command Command::STOP(1ul << 27, "");
 const Command Command::SHIFT(1uL << 28, "");
-
+const Command Command::WFORWARD(1uL << 29, "Wingman forward thrust");
+const Command Command::WLEFT(1uL << 30, "Wingman turn left");
+const Command Command::WRIGHT(1uL << 31, "Wingman turn right");
+const Command Command::WBACK(1uLL << 32, "Wingman reverse");
+const Command Command::WPRIMARY(1uLL << 33, "Wingman fire primary weapon");
+const Command Command::WSECONDARY(1uLL << 34, "Wingman fire secondary weapon");
+const Command Command::WSELECT(1uLL << 35, "Wingman select secondary weapon");
+const Command Command::WJUMP(1uLL << 36, "Wingman initiate hyperspace jump");
+const Command Command::WAFTERBURNER(1uLL << 37, "Wingman fire afterburner");
```

Comment:
I would expect one command class instance for the first player and another command instance for each additional wingman.

Reply:
So my understanding is that right now there is a single `activeCommands` object that encompasses all of the keys that have been pushed/held. I am having some trouble thinking about where this list of commands would be stored and how it would be organized. Are you thinking that any place that `activeCommands` should be an array or map or something that we pass into any place that currently takes one `Command` instance today and that CalculateStep() UpdateKeys() etc iterate over the active command list and update ships accordingly?

Reply:
I didn't check, but I expect the AI ships to just fire any weapons they want to fire. A computer doesn't need to select weapons, such selection is only needed for players that have a limited set of input keys. We could argue why we only allow the player to fire 1 secondary weapon at a time, but that is for a different time and a different PR.

Giving different players their own weapon selection would probably require a split-up of the PlayerInfo class. It currently handles a lot of generic world/universe data and a lot of data that would be per-player.

Reply:
The command class is used for many different things; keyboard input from the player(s), control inputs to ships, tracking of which weapons to fire.
I expect some refactoring to be needed to read keyboard input in a different way and generate separate command class instances per player.